.github/workflows/external-pkcs11.yaml

```diff
@@ -0,0 +1,114 @@
+name: Tests of external pkcs11 modules
+
+on:
+  pull_request:
+    paths:
+      - '**.c'
+      - '**.h'
+      - '**.sh'
+      - .github/workflows/fedora.yml
+      - '**.am'
+      - doc/**
+      - configure.ac
+  push:
+
+permissions:
+  contents: read  #  to fetch code (actions/checkout)
+
+env:
+  BASE_DEPS: |
+    build-essential docbook-xsl xsltproc gengetopt help2man pcscd check
+    pcsc-tools libtool make autoconf autoconf-archive automake pkg-config
+    git xxd openssl valgrind socat gawk
+    libglib2.0-dev libnss3-dev gnutls-bin libusb-dev libudev-dev flex
+    libnss3-tools
+    libpcsclite-dev libcmocka-dev libssl-dev zlib1g-dev libreadline-dev softhsm2
+  KRYOPTIC: https://github.com/latchset/kryoptic.git
+
+jobs:
+  #######################
+  ## pkcs11-tool tests ##
+  #######################
+
+  test-kryoptic:
+    runs-on: ubuntu-24.04
+    container:
+      image: fedora:latest
+    env:
+      SOURCE_PATH: ./
+      BUILD_PATH: ./
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - name: Setup Fedora
+        run: .github/setup-fedora.sh kryoptic
+      - name: Install OpenSC
+        run: .github/build.sh
+      - name: Setup Kryoptic
+        run: .github/setup-kryoptic.sh
+      - name: Test Sign & Verify with Kryoptic
+        run: tests/test-pkcs11-tool-sign-verify.sh kryoptic
+      - name: Test Import with Kryoptic
+        run: tests/test-pkcs11-tool-import.sh kryoptic kryoptic
+      - name: Test Allowed Mechanisms with Kryoptic
+        run: tests/test-pkcs11-tool-allowed-mechanisms.sh kryoptic
+      - name: Test pkcs11-tool test with Kryoptic
+        run: tests/test-pkcs11-tool-test.sh kryoptic
```

Comment:
I would rather have this in separate script that could be executed locally easily